--- origsrc/gnome-flashback-3.22.0/configure.ac	2016-10-06 09:34:00.000000000 -0500
+++ src/gnome-flashback-3.22.0/configure.ac	2017-03-07 14:09:24.892154800 -0600
@@ -134,7 +134,8 @@ PKG_CHECK_MODULES([AUTOMOUNT_MANAGER], [
 PKG_CHECK_MODULES([BLUETOOTH_APPLET], [
   gtk+-3.0 >= $GTK_REQUIRED
   gnome-bluetooth-1.0
-])
+],[BLUETOOTH=yes],[BLUETOOTH=no])
+AM_CONDITIONAL([WITH_BLUETOOTH],[test "x$BLUETOOTH" = "xyes"])
 
 PKG_CHECK_MODULES([COMMON], [
   gtk+-3.0 >= $GTK_REQUIRED
@@ -150,7 +151,6 @@ PKG_CHECK_MODULES([DISPLAY_CONFIG], [
   gtk+-3.0 >= $GTK_REQUIRED
   glib-2.0 >= $GLIB_REQUIRED
   gnome-desktop-3.0 >= $LIBGNOME_DESKTOP_REQUIRED
-  upower-glib >= $UPOWER_GLIB_REQUIRED
   x11
   xext
   xrandr
@@ -197,12 +197,14 @@ PKG_CHECK_MODULES([POLKIT], [
   glib-2.0 >= $GLIB_REQUIRED
   polkit-agent-1 >= $POLKIT_AGENT_REQUIRED
   polkit-gobject-1 >= $POLKIT_GOBJECT_REQUIRED
-])
+],[POLKIT=yes],[POLKIT=no])
+AM_CONDITIONAL([WITH_POLKIT],[test "x$POLKIT" = "xyes"])
 
 PKG_CHECK_MODULES([POWER_APPLET], [
   gtk+-3.0 >= $GTK_REQUIRED
   upower-glib
-])
+],[UPOWER=yes],[UPOWER=no])
+AM_CONDITIONAL([WITH_UPOWER],[test "x$UPOWER" = "xyes"])
 
 PKG_CHECK_MODULES([SCREENCAST], [
   gtk+-3.0 >= $GTK_REQUIRED
--- origsrc/gnome-flashback-3.22.0/gnome-flashback/Makefile.am	2016-10-05 09:21:01.000000000 -0500
+++ src/gnome-flashback-3.22.0/gnome-flashback/Makefile.am	2017-03-07 14:10:43.196659800 -0600
@@ -3,7 +3,6 @@ NULL =
 SUBDIRS = \
 	libaudio-device-selection \
 	libautomount-manager \
-	libbluetooth-applet \
 	libcommon \
 	libdesktop-background \
 	libdisplay-config \
@@ -12,8 +11,6 @@ SUBDIRS = \
 	libinput-settings \
 	libinput-sources \
 	libnotifications \
-	libpolkit \
-	libpower-applet \
 	libscreencast \
 	libscreensaver \
 	libscreenshot \
@@ -22,6 +19,16 @@ SUBDIRS = \
 	libworkarounds \
 	$(NULL)
 
+if WITH_BLUETOOTH
+SUBDIRS += libbluetooth-applet
+endif
+if WITH_POLKIT
+SUBDIRS += libpolkit
+endif
+if WITH_UPOWER
+SUBDIRS += libpower-applet
+endif
+
 bin_PROGRAMS = \
 	gnome-flashback \
 	$(NULL)
@@ -52,7 +59,6 @@ gnome_flashback_LDADD = \
 	$(GNOME_FLASHBACK_LIBS) \
 	$(top_builddir)/gnome-flashback/libaudio-device-selection/libaudio-device-selection.la \
 	$(top_builddir)/gnome-flashback/libautomount-manager/libautomount-manager.la \
-	$(top_builddir)/gnome-flashback/libbluetooth-applet/libbluetooth-applet.la \
 	$(top_builddir)/gnome-flashback/libdesktop-background/libdesktop-background.la \
 	$(top_builddir)/gnome-flashback/libdisplay-config/libdisplay-config.la \
 	$(top_builddir)/gnome-flashback/libend-session-dialog/libend-session-dialog.la \
@@ -60,8 +66,6 @@ gnome_flashback_LDADD = \
 	$(top_builddir)/gnome-flashback/libinput-settings/libinput-settings.la \
 	$(top_builddir)/gnome-flashback/libinput-sources/libinput-sources.la \
 	$(top_builddir)/gnome-flashback/libnotifications/libnotifications.la \
-	$(top_builddir)/gnome-flashback/libpolkit/libpolkit.la \
-	$(top_builddir)/gnome-flashback/libpower-applet/libpower-applet.la \
 	$(top_builddir)/gnome-flashback/libscreencast/libscreencast.la \
 	$(top_builddir)/gnome-flashback/libscreensaver/libscreensaver.la \
 	$(top_builddir)/gnome-flashback/libscreenshot/libscreenshot.la \
@@ -70,6 +74,19 @@ gnome_flashback_LDADD = \
 	$(top_builddir)/gnome-flashback/libworkarounds/libworkarounds.la \
 	$(NULL)
 
+if WITH_BLUETOOTH
+gnome_flashback_LDADD += \
+	$(top_builddir)/gnome-flashback/libbluetooth-applet/libbluetooth-applet.la
+endif
+if WITH_POLKIT
+gnome_flashback_LDADD += \
+	$(top_builddir)/gnome-flashback/libpolkit/libpolkit.la
+endif
+if WITH_UPOWER
+gnome_flashback_LDADD += \
+	$(top_builddir)/gnome-flashback/libpower-applet/libpower-applet.la
+endif
+
 gf-resources.c: flashback.gresource.xml $(shell $(GLIB_COMPILE_RESOURCES) --sourcedir=$(srcdir) --generate-dependencies $(srcdir)/flashback.gresource.xml)
 	$(AM_V_GEN)$(GLIB_COMPILE_RESOURCES) --target=$@ --sourcedir=$(srcdir) --generate --c-name gf $<
 
--- origsrc/gnome-flashback-3.22.0/gnome-flashback/gf-application.c	2016-10-05 09:21:01.000000000 -0500
+++ src/gnome-flashback-3.22.0/gnome-flashback/gf-application.c	2017-03-07 14:12:13.485093600 -0600
@@ -156,16 +156,22 @@ settings_changed (GSettings   *settings,
   SETTING_CHANGED (automount, "automount-manager", gsd_automount_manager_new)
   SETTING_CHANGED (config, "display-config", flashback_display_config_new)
   SETTING_CHANGED (idle_monitor, "idle-monitor", flashback_idle_monitor_new)
+#if 0
   SETTING_CHANGED (polkit, "polkit", flashback_polkit_new)
+#endif
   SETTING_CHANGED (shell, "shell", flashback_shell_new)
   SETTING_CHANGED (audio_device_selection, "audio-device-selection", gf_audio_device_selection_new)
+#if 0
   SETTING_CHANGED (bluetooth, "bluetooth-applet", gf_bluetooth_applet_new)
+#endif
   SETTING_CHANGED (background, "desktop-background", gf_desktop_background_new)
   SETTING_CHANGED (dialog, "end-session-dialog", gf_end_session_dialog_new)
   SETTING_CHANGED (input_settings, "input-settings", gf_input_settings_new)
   SETTING_CHANGED (input_sources, "input-sources", gf_input_sources_new)
   SETTING_CHANGED (notifications, "notifications", gf_notifications_new)
+#if 0
   SETTING_CHANGED (power, "power-applet", gf_power_applet_new)
+#endif
   SETTING_CHANGED (screencast, "screencast", gf_screencast_new)
   SETTING_CHANGED (screensaver, "screensaver", gf_screensaver_new)
   SETTING_CHANGED (screenshot, "screenshot", gf_screenshot_new)
@@ -217,15 +223,21 @@ gf_application_dispose (GObject *object)
   g_clear_object (&application->automount);
   g_clear_object (&application->config);
   g_clear_object (&application->idle_monitor);
+#if 0
   g_clear_object (&application->polkit);
+#endif
   g_clear_object (&application->shell);
   g_clear_object (&application->audio_device_selection);
+#if 0
   g_clear_object (&application->bluetooth);
+#endif
   g_clear_object (&application->background);
   g_clear_object (&application->dialog);
   g_clear_object (&application->input_sources);
   g_clear_object (&application->notifications);
+#if 0
   g_clear_object (&application->power);
+#endif
   g_clear_object (&application->screencast);
   g_clear_object (&application->screenshot);
   g_clear_object (&application->screensaver);
--- origsrc/gnome-flashback-3.22.0/gnome-flashback/libdisplay-config/flashback-monitor-config.c	2016-10-05 09:21:01.000000000 -0500
+++ src/gnome-flashback-3.22.0/gnome-flashback/libdisplay-config/flashback-monitor-config.c	2017-03-07 14:13:52.028956800 -0600
@@ -33,7 +33,6 @@
 
 #include <config.h>
 #include <string.h>
-#include <libupower-glib/upower.h>
 #include "flashback-monitor-config.h"
 
 /* These structures represent the intended/persistent configuration,
@@ -117,7 +116,6 @@ struct _FlashbackMonitorConfig
   GFile                   *system_file;
   GCancellable            *save_cancellable;
 
-  UpClient                *up_client;
   gboolean                 lid_is_closed;
 
   FlashbackMonitorManager *manager;
@@ -749,6 +747,7 @@ turn_off_laptop_display (FlashbackMonito
   config->current_is_for_laptop_lid = TRUE;
 }
 
+#if 0
 static void
 power_client_changed_cb (UpClient   *client,
                          GParamSpec *pspec,
@@ -770,6 +769,7 @@ power_client_changed_cb (UpClient   *cli
         flashback_monitor_config_restore_previous (config);
     }
 }
+#endif
 
 static void
 handle_start_element (GMarkupParseContext  *context,
@@ -1929,11 +1929,7 @@ flashback_monitor_config_init (Flashback
       g_free (path);
     }
 
-  config->up_client = up_client_new ();
-  config->lid_is_closed = up_client_get_lid_is_closed (config->up_client);
-
-  g_signal_connect_object (config->up_client, "notify::lid-is-closed",
-                           G_CALLBACK (power_client_changed_cb), config, 0);
+  config->lid_is_closed = FALSE;
 }
 
 FlashbackMonitorConfig *
