--- origsrc/gnome-flashback-3.26.0/backends/gf-monitor-manager-private.h	2017-09-29 06:48:07.000000000 -0500
+++ src/gnome-flashback-3.26.0/backends/gf-monitor-manager-private.h	2018-02-14 23:43:09.037879700 -0600
@@ -26,7 +26,9 @@
 #define GF_MONITOR_MANAGER_PRIVATE_H
 
 #include <libgnome-desktop/gnome-pnp-ids.h>
+#if 0
 #include <libupower-glib/upower.h>
+#endif
 
 #include "gf-backend-private.h"
 #include "gf-dbus-display-config.h"
@@ -81,7 +83,11 @@ struct _GfMonitorManager
   GfMonitorConfigManager      *config_manager;
 
   GnomePnpIds                 *pnp_ids;
+#if 0
   UpClient                    *up_client;
+#else
+  void                        *up_client;
+#endif
 
   GfMonitorSwitchConfigType    current_switch_config;
 };
--- origsrc/gnome-flashback-3.26.0/backends/gf-monitor-manager.c	2017-10-02 17:15:56.000000000 -0500
+++ src/gnome-flashback-3.26.0/backends/gf-monitor-manager.c	2018-02-14 23:45:15.631018900 -0600
@@ -1935,9 +1935,12 @@ gf_monitor_manager_real_is_lid_closed (G
   if (!manager->up_client)
     return FALSE;
 
+#if 0
   return up_client_get_lid_is_closed (manager->up_client);
+#endif
 }
 
+#if 0
 static void
 lid_is_closed_changed (UpClient   *client,
                        GParamSpec *pspec,
@@ -1947,6 +1950,7 @@ lid_is_closed_changed (UpClient   *clien
 
   gf_monitor_manager_ensure_configured (manager);
 }
+#endif
 
 static void
 gf_monitor_manager_constructed (GObject *object)
@@ -1962,6 +1966,7 @@ gf_monitor_manager_constructed (GObject
 
   G_OBJECT_CLASS (gf_monitor_manager_parent_class)->constructed (object);
 
+#if 0
   if (manager_class->is_lid_closed == gf_monitor_manager_real_is_lid_closed)
     {
       manager->up_client = up_client_new ();
@@ -1969,6 +1974,7 @@ gf_monitor_manager_constructed (GObject
       g_signal_connect_object (manager->up_client, "notify::lid-is-closed",
                                G_CALLBACK (lid_is_closed_changed), manager, 0);
     }
+#endif
 
   g_signal_connect_object (manager, "notify::power-save-mode",
                            G_CALLBACK (power_save_mode_changed), manager, 0);
@@ -2013,7 +2019,9 @@ gf_monitor_manager_dispose (GObject *obj
     }
 
   g_clear_object (&manager->config_manager);
+#if 0
   g_clear_object (&manager->up_client);
+#endif
 
   priv->backend = NULL;
 
--- origsrc/gnome-flashback-3.26.0/configure.ac	2017-10-05 06:18:19.000000000 -0500
+++ src/gnome-flashback-3.26.0/configure.ac	2018-02-14 23:35:50.564585300 -0600
@@ -104,7 +104,6 @@ PKG_CHECK_MODULES([BACKENDS], [
   gio-unix-2.0 >= $GLIB_REQUIRED
   glib-2.0 >= $GLIB_REQUIRED
   gnome-desktop-3.0 >= $LIBGNOME_DESKTOP_REQUIRED
-  upower-glib >= $UPOWER_GLIB_REQUIRED
   xcb-randr
   xrandr >= $XRANDR_REQUIRED
   x11-xcb
@@ -128,7 +127,8 @@ PKG_CHECK_MODULES([AUTOMOUNT_MANAGER], [
 PKG_CHECK_MODULES([BLUETOOTH_APPLET], [
   gtk+-3.0 >= $GTK_REQUIRED
   gnome-bluetooth-1.0
-])
+],[BLUETOOTH=yes],[BLUETOOTH=no])
+AM_CONDITIONAL([WITH_BLUETOOTH],[test "x$BLUETOOTH" = "xyes"])
 
 PKG_CHECK_MODULES([COMMON], [
   gtk+-3.0 >= $GTK_REQUIRED
@@ -179,12 +179,14 @@ PKG_CHECK_MODULES([POLKIT], [
   glib-2.0 >= $GLIB_REQUIRED
   polkit-agent-1 >= $POLKIT_AGENT_REQUIRED
   polkit-gobject-1 >= $POLKIT_GOBJECT_REQUIRED
-])
+],[POLKIT=yes],[POLKIT=no])
+AM_CONDITIONAL([WITH_POLKIT],[test "x$POLKIT" = "xyes"])
 
 PKG_CHECK_MODULES([POWER_APPLET], [
   gtk+-3.0 >= $GTK_REQUIRED
   upower-glib
-])
+],[UPOWER=yes],[UPOWER=no])
+AM_CONDITIONAL([WITH_UPOWER],[test "x$UPOWER" = "xyes"])
 
 PKG_CHECK_MODULES([SCREENCAST], [
   gtk+-3.0 >= $GTK_REQUIRED
--- origsrc/gnome-flashback-3.26.0/data/sessions/Makefile.am	2017-08-20 09:46:03.000000000 -0500
+++ src/gnome-flashback-3.26.0/data/sessions/Makefile.am	2018-02-14 22:02:06.572761000 -0600
@@ -11,13 +11,10 @@ required_components_gsd = \
 	org.gnome.SettingsDaemon.A11yKeyboard \
 	org.gnome.SettingsDaemon.A11ySettings \
 	org.gnome.SettingsDaemon.Clipboard \
-	org.gnome.SettingsDaemon.Color \
-	org.gnome.SettingsDaemon.Datetime \
 	org.gnome.SettingsDaemon.Housekeeping \
 	org.gnome.SettingsDaemon.Keyboard \
 	org.gnome.SettingsDaemon.MediaKeys \
 	org.gnome.SettingsDaemon.Mouse \
-	org.gnome.SettingsDaemon.Power \
 	org.gnome.SettingsDaemon.PrintNotifications \
 	org.gnome.SettingsDaemon.Rfkill \
 	org.gnome.SettingsDaemon.ScreensaverProxy \
--- origsrc/gnome-flashback-3.26.0/gnome-flashback/Makefile.am	2017-09-29 06:23:10.000000000 -0500
+++ src/gnome-flashback-3.26.0/gnome-flashback/Makefile.am	2018-02-14 22:02:59.866886000 -0600
@@ -3,7 +3,6 @@ NULL =
 SUBDIRS = \
 	libaudio-device-selection \
 	libautomount-manager \
-	libbluetooth-applet \
 	libcommon \
 	libdesktop-background \
 	libend-session-dialog \
@@ -11,8 +10,6 @@ SUBDIRS = \
 	libinput-settings \
 	libinput-sources \
 	libnotifications \
-	libpolkit \
-	libpower-applet \
 	libscreencast \
 	libscreensaver \
 	libscreenshot \
@@ -22,6 +19,16 @@ SUBDIRS = \
 	libworkarounds \
 	$(NULL)
 
+if WITH_BLUETOOTH
+SUBDIRS += libbluetooth-applet
+endif
+if WITH_POLKIT
+SUBDIRS += libpolkit
+endif
+if WITH_UPOWER
+SUBDIRS += libpower-applet
+endif
+
 bin_PROGRAMS = \
 	gnome-flashback \
 	$(NULL)
@@ -62,15 +69,12 @@ gnome_flashback_LDADD = \
 	$(top_builddir)/backends/libbackends.la \
 	$(top_builddir)/gnome-flashback/libaudio-device-selection/libaudio-device-selection.la \
 	$(top_builddir)/gnome-flashback/libautomount-manager/libautomount-manager.la \
-	$(top_builddir)/gnome-flashback/libbluetooth-applet/libbluetooth-applet.la \
 	$(top_builddir)/gnome-flashback/libdesktop-background/libdesktop-background.la \
 	$(top_builddir)/gnome-flashback/libend-session-dialog/libend-session-dialog.la \
 	$(top_builddir)/gnome-flashback/libidle-monitor/libidle-monitor.la \
 	$(top_builddir)/gnome-flashback/libinput-settings/libinput-settings.la \
 	$(top_builddir)/gnome-flashback/libinput-sources/libinput-sources.la \
 	$(top_builddir)/gnome-flashback/libnotifications/libnotifications.la \
-	$(top_builddir)/gnome-flashback/libpolkit/libpolkit.la \
-	$(top_builddir)/gnome-flashback/libpower-applet/libpower-applet.la \
 	$(top_builddir)/gnome-flashback/libscreencast/libscreencast.la \
 	$(top_builddir)/gnome-flashback/libscreensaver/libscreensaver.la \
 	$(top_builddir)/gnome-flashback/libscreenshot/libscreenshot.la \
@@ -80,6 +84,19 @@ gnome_flashback_LDADD = \
 	$(top_builddir)/gnome-flashback/libworkarounds/libworkarounds.la \
 	$(NULL)
 
+if WITH_BLUETOOTH
+gnome_flashback_LDADD += \
+	$(top_builddir)/gnome-flashback/libbluetooth-applet/libbluetooth-applet.la
+endif
+if WITH_POLKIT
+gnome_flashback_LDADD += \
+	$(top_builddir)/gnome-flashback/libpolkit/libpolkit.la
+endif
+if WITH_UPOWER
+gnome_flashback_LDADD += \
+	$(top_builddir)/gnome-flashback/libpower-applet/libpower-applet.la
+endif
+
 gf-resources.c: flashback.gresource.xml $(shell $(GLIB_COMPILE_RESOURCES) --sourcedir=$(srcdir) --generate-dependencies $(srcdir)/flashback.gresource.xml)
 	$(AM_V_GEN)$(GLIB_COMPILE_RESOURCES) --target=$@ --sourcedir=$(srcdir) --generate --c-name gf $<
 
--- origsrc/gnome-flashback-3.26.0/gnome-flashback/gf-application.c	2017-09-29 06:24:28.000000000 -0500
+++ src/gnome-flashback-3.26.0/gnome-flashback/gf-application.c	2018-02-14 22:02:59.866886000 -0600
@@ -155,16 +155,22 @@ settings_changed (GSettings   *settings,
 
   SETTING_CHANGED (automount, "automount-manager", gsd_automount_manager_new)
   SETTING_CHANGED (idle_monitor, "idle-monitor", flashback_idle_monitor_new)
+#if 0
   SETTING_CHANGED (polkit, "polkit", flashback_polkit_new)
+#endif
   SETTING_CHANGED (shell, "shell", flashback_shell_new)
   SETTING_CHANGED (audio_device_selection, "audio-device-selection", gf_audio_device_selection_new)
+#if 0
   SETTING_CHANGED (bluetooth, "bluetooth-applet", gf_bluetooth_applet_new)
+#endif
   SETTING_CHANGED (background, "desktop-background", gf_desktop_background_new)
   SETTING_CHANGED (dialog, "end-session-dialog", gf_end_session_dialog_new)
   SETTING_CHANGED (input_settings, "input-settings", gf_input_settings_new)
   SETTING_CHANGED (input_sources, "input-sources", gf_input_sources_new)
   SETTING_CHANGED (notifications, "notifications", gf_notifications_new)
+#if 0
   SETTING_CHANGED (power, "power-applet", gf_power_applet_new)
+#endif
   SETTING_CHANGED (screencast, "screencast", gf_screencast_new)
   SETTING_CHANGED (screensaver, "screensaver", gf_screensaver_new)
   SETTING_CHANGED (screenshot, "screenshot", gf_screenshot_new)
@@ -201,15 +207,21 @@ gf_application_dispose (GObject *object)
 
   g_clear_object (&application->automount);
   g_clear_object (&application->idle_monitor);
+#if 0
   g_clear_object (&application->polkit);
+#endif
   g_clear_object (&application->shell);
   g_clear_object (&application->audio_device_selection);
+#if 0
   g_clear_object (&application->bluetooth);
+#endif
   g_clear_object (&application->background);
   g_clear_object (&application->dialog);
   g_clear_object (&application->input_sources);
   g_clear_object (&application->notifications);
+#if 0
   g_clear_object (&application->power);
+#endif
   g_clear_object (&application->screencast);
   g_clear_object (&application->screenshot);
   g_clear_object (&application->screensaver);
